resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/HttpClientUtil.scala

```diff
@@ -57,11 +58,23 @@ private[spark] object HttpClientUtil {
         response
       }
     }
+    val defaultErrorDecoder = new ErrorDecoder.Default
+    val alwaysRetryErrorDecoder = new ErrorDecoder {
+      override def decode(methodKey: String, response: Response): Exception = {
+        defaultErrorDecoder.decode(methodKey, response) match {
```

Comment:
The Ingress code path may return errors that are 404s when the load balancer (Nginx in the case of Minikube) has not set up the proxy yet. These kinds of errors aren't considered retryable by default.